tests/testthat/test-grade_this.R

```diff
@@ -17,3 +17,29 @@ test_that("grade_this() can return early", {
     })(mock_this_exercise(1))
   )
 })
+
+test_that("grade_this() can find things in the parent of the check env", {
+  # inspired by tblcheck::grade_this_table()
+  ex <- mock_this_exercise(
+    .user_code = "from_user <- TRUE",
+    setup_exercise = "from_setup <- TRUE"
+  )
+
+  new_parent <- rlang::new_environment(
+    list(from_parent = TRUE),
+    parent = rlang::env_parent(ex)
+  )
+  rlang::env_poke_parent(ex, new_parent)
+
+  found <- grade_this(
+    pass(data = list(
+      from_parent = from_parent,
+      from_setup = from_setup,
+      from_user = .envir_result$from_user
+    ))
+  )(ex)$data
+
+  expect_true(found$from_parent)
+  expect_true(found$from_setup)
+  expect_true(found$from_user)
+})
```

Comment:
This is great, I was planning to add tests within `tblcheck` once this merged, but it's even nicer to have it all handled here!

Reply:
Thanks! Hopefully it'll be an early warning system